airflow/providers/postgres/hooks/postgres.py

```diff
@@ -67,10 +68,18 @@ class PostgresHook(DbApiHook):
     supports_autocommit = True
 
     def __init__(self, *args, **kwargs) -> None:
+        if 'schema' in kwargs:
+            warnings.warn(
+                'The "schema" arg has been renamed to "database" as it contained the database name.'
+                'Please use "database" to set the database name.',
+                DeprecationWarning,
+                stacklevel=2,
+            )
+            kwargs['database'] = kwargs['schema']
         super().__init__(*args, **kwargs)
         self.connection: Connection | None = kwargs.pop("connection", None)
         self.conn: connection = None
-        self.schema: str | None = kwargs.pop("schema", None)
+        self.database: str | None = kwargs.pop("database", None)
```

Comment:
Thought: Should we add `schema` as property and raise deprecation warning for folks relying on `PostgresHook().schema`? Or we rather give them `AttributeError` as it is wrong anyways?

Reply:
I don't have a strong opinion on this. @ashb @potiuk maybe?

Reply:
I don't think we can even make it deprecated really:

This is in DbAPIHook:

```    def get_conn(self):
        """Returns a connection object"""
        db = self.get_connection(getattr(self, self.conn_name_attr))
        return self.connector.connect(host=db.host, port=db.port, username=db.login, schema=db.schema)
```

Reply:
`db.schema` there refers to the connection schema - not the hook schema attribute.

Reply:
Oh right yes.

In which case I think it's still worth having a deprecation warning property.

Reply:
Okay, will add it.

Reply:
Added in 4a7ae31